unstructured/ingest/interfaces2.py

```diff
@@ -0,0 +1,446 @@
+"""Defines Abstract Base Classes (ABC's) core to batch processing documents
+through Unstructured."""
+
+import functools
+import json
+import os
+import typing as t
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from datetime import datetime
+from pathlib import Path
+
+import requests
+from dataclasses_json import DataClassJsonMixin
+
+from unstructured.documents.elements import DataSourceMetadata
+from unstructured.ingest.error import PartitionError, SourceConnectionError
+from unstructured.ingest.logger import logger
+from unstructured.partition.auto import partition
+from unstructured.staging.base import convert_to_dict
+
+
+@dataclass
+class BaseSessionHandle(ABC):
+    """Abstract Base Class for sharing resources that are local to an individual process.
+    e.g., a connection for making a request for fetching documents."""
+
+
+class BaseConfig(DataClassJsonMixin, ABC):
+    pass
+
+
+@dataclass
+class PartitionConfig(BaseConfig):
+    # where to write structured data outputs
+    output_dir: str = "structured-output"
+    num_processes: int = 2
+    max_docs: t.Optional[int] = None
+    pdf_infer_table_structure: bool = False
+    strategy: str = "auto"
+    reprocess: bool = False
+    ocr_languages: str = "eng"
+    encoding: t.Optional[str] = None
+    fields_include: t.List[str] = field(
+        default_factory=lambda: ["element_id", "text", "type", "metadata"],
+    )
+    flatten_metadata: bool = False
+    metadata_exclude: t.List[str] = field(default_factory=list)
+    metadata_include: t.List[str] = field(default_factory=list)
+    partition_endpoint: t.Optional[str] = None
+    api_key: t.Optional[str] = None
+
+    @staticmethod
+    def get_schema() -> dict:
```

Comment:
is the idea that we would want to validate against this in the future? allow users to inspect what's needed?

Reply:
needed for adding the s3 writer? or just adding as something we should do across the board?

Reply:
Removed this from the base config class, forgot to remove the implementation. This is now gone. I had cherry picked this from the other PR I had out but don't need if for this specific one.

Reply:
cool cool, thanks